Makefile

```diff
@@ -88,21 +88,22 @@ APPLICATION_VERSION_DEBIAN = $(shell echo $(APPLICATION_VERSION) | tr "-" "~")
 # Rules
 # ---------------------------------------------------------------------
 
-release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app:
-	./scripts/unix/electron-create-resources-app.sh -s . -f "$(APPLICATION_FILES)" -o $@
+release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/node_modules: package.json npm-shrinkwrap.json
+	./scripts/unix/dependencies-npm.sh -p -r "$(TARGET_ARCH)" -v "$(ELECTRON_VERSION)" -x $(dir $@) -t electron
 
-release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app/node_modules:\
-	release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app
-	./scripts/unix/dependencies-npm.sh -p -r "$(TARGET_ARCH)" -v "$(ELECTRON_VERSION)" -x $< -t electron
+release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/bower_components: bower.json
+	./scripts/unix/dependencies-bower.sh -p -x $(dir $@)
 
-release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app/bower_components:\
-	release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app
-	./scripts/unix/dependencies-bower.sh -p -x $<
+release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app: \
+	release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/node_modules \
```

Comment:
Another problem with having directories as Makefile dependencies (that I've just discovered) is that if you have a script which creates the directory and then executes some process(es) to update the files inside it, and the update-process fails, then the next time you run `make` it'll see that the directory already exists, assume everything is up-to-date, and not do anything.
So a better approach would be to `touch` a "marker file" _after_ the update process has finished (e.g. `.npm_dependencies_installed`), and use _that_ as the dependency, so that if it fails `make` knows that it needs to try running the update-process again next time.